From 1abf1924f1c3ccaa58758e3415eb6150f6f0d1f3 Mon Sep 17 00:00:00 2001
From: YukonAppleGeek <yukonvinecki@gmail.com>
Date: Sat, 29 Jun 2013 22:03:21 -0700
Subject: [PATCH] Add BlockFallEvent


diff --git a/src/main/java/net/minecraft/server/BlockFalling.java b/src/main/java/net/minecraft/server/BlockFalling.java
index 1d5fe2f..020924b 100644
--- a/src/main/java/net/minecraft/server/BlockFalling.java
+++ b/src/main/java/net/minecraft/server/BlockFalling.java
@@ -44,6 +44,16 @@ public class BlockFalling extends Block {
                 if (!var1.isStatic) {
                     EntityFallingBlock var9 = new EntityFallingBlock(var1, (double)((float)var2 + 0.5F), (double)((float)var3 + 0.5F), (double)((float)var4 + 0.5F), this, var1.getData(var2, var3, var4));
                     this.a(var9);
+
+                    org.bukkit.block.Block block = world.getWorld().getBlockAt(i, j, k);
+
+                    BlockFallEvent blockFallEvent = new BlockFallEvent(block, (org.bukkit.entity.FallingBlock) entityfallingblock.getBukkitEntity());
+                    world.getServer().getPluginManager().callEvent(blockFallEvent);
+                    
+                    if (blockFallEvent.isCancelled()) {
+                        return;
+                    }
+
                     var1.addEntity(var9);
                 }
             } else {
-- 
1.8.3.2

