From d23305feb9cfb113f97a40900b611a4b3c82a944 Mon Sep 17 00:00:00 2001
From: Steve Anton <anxuiz.nx@gmail.com>
Date: Fri, 5 Oct 2012 12:00:26 -0700
Subject: [PATCH] Call BlockFormEvent for obsidian and cobblestone formation
 due to lava / water interacting.


diff --git a/src/main/java/net/minecraft/server/BlockFluids.java b/src/main/java/net/minecraft/server/BlockFluids.java
index c28eb62..6f493fe 100644
--- a/src/main/java/net/minecraft/server/BlockFluids.java
+++ b/src/main/java/net/minecraft/server/BlockFluids.java
@@ -216,14 +216,26 @@ public abstract class BlockFluids extends Block {
 
                 if (var5) {
                     int var6 = var1.getData(var2, var3, var4);
+                    Block formedBlock = null;
 
                     if (var6 == 0) {
-                        var1.setTypeUpdate(var2, var3, var4, Blocks.OBSIDIAN);
+                        formedBlock = Blocks.OBSIDIAN;
                     } else if (var6 <= 4) {
-                        var1.setTypeUpdate(var2, var3, var4, Blocks.COBBLESTONE);
+                        formedBlock = Blocks.COBBLESTONE;
                     }
 
-                    this.fizz(var1, var2, var3, var4);
+                    if (formedBlock != null) {
+                        org.bukkit.block.Block block = var1.getWorld().getBlockAt(var2, var3, var4);
+                        org.bukkit.block.BlockState newState = block.getState();
+                        newState.setType(org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(formedBlock));
+                        org.bukkit.event.block.BlockFormEvent event = new org.bukkit.event.block.BlockFormEvent(block, newState);
+                        var1.getServer().getPluginManager().callEvent(event);
+
+                        if (!event.isCancelled()) {
+                            var1.setTypeUpdate(var2, var3, var4, formedBlock);
+                            this.fizz(var1, var2, var3, var4);
+                        }
+                    }
                 }
             }
         }
-- 
1.8.3.2

