From 3ea27db53a45734d1ed573aae1f98713ca7b416b Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Sun, 16 Jun 2013 17:03:07 -0500
Subject: [PATCH] Add getting and setting of entity count


diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index dc0abc5..845a582 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -38,7 +38,7 @@ public abstract class Entity {
     }
     // CraftBukkit end
 
-    private static int entityCount;
+    public static int entityCount;
     private int id;
     public double k;
     public boolean l;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a02fe26..7473c8b 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1470,4 +1470,12 @@ public final class CraftServer implements Server {
     public UnsafeValues getUnsafe() {
         return CraftMagicNumbers.INSTANCE;
     }
+
+    public int getEntityCount() {
+        return net.minecraft.server.Entity.entityCount;
+    }
+
+    public void setEntityCount(int count) {
+        net.minecraft.server.Entity.entityCount = count;
+    }
 }
-- 
1.8.3.2

