From 02c5ca71ac03ecc928f712e0a43aaf7b30075ba7 Mon Sep 17 00:00:00 2001
From: Chad Waters <authorblues@gmail.com>
Date: Thu, 2 Aug 2012 16:47:02 -0700
Subject: [PATCH] Fire ExplosionPrimeEvent properly. Fixes BUKKIT-770


diff --git a/src/main/java/net/minecraft/server/BlockTNT.java b/src/main/java/net/minecraft/server/BlockTNT.java
index d690ebe..e913d43 100644
--- a/src/main/java/net/minecraft/server/BlockTNT.java
+++ b/src/main/java/net/minecraft/server/BlockTNT.java
@@ -14,6 +14,7 @@ import net.minecraft.server.Material;
 import net.minecraft.server.World;
 
 public class BlockTNT extends Block {
+    boolean removeBlock = false;
 
     public BlockTNT() {
         super(Material.TNT);
@@ -43,8 +44,15 @@ public class BlockTNT extends Block {
     public void wasExploded(World var1, int var2, int var3, int var4, Explosion var5) {
         if (!var1.isStatic) {
             EntityTNTPrimed var6 = new EntityTNTPrimed(var1, (double)((float)var2 + 0.5F), (double)((float)var3 + 0.5F), (double)((float)var4 + 0.5F), var5.c());
-            var6.fuseTicks = var1.random.nextInt(var6.fuseTicks / 4) + var6.fuseTicks / 8;
-            var1.addEntity(var6);
+            // CraftBukkit start - allow exploded TNT to cancel priming
+            org.bukkit.event.entity.ExplosionPrimeEvent event = new org.bukkit.event.entity.ExplosionPrimeEvent((org.bukkit.entity.Explosive) var6.getBukkitEntity());
+            var1.getServer().getPluginManager().callEvent(event);
+
+            if (!event.isCancelled()) {
+                var6.fuseTicks = var1.random.nextInt(var6.fuseTicks / 4) + var6.fuseTicks / 8;
+                var1.addEntity(var6);
+            }
+            // CraftBukkit end
         }
     }
 
@@ -53,11 +61,22 @@ public class BlockTNT extends Block {
     }
 
     public void a(World var1, int var2, int var3, int var4, int var5, EntityLiving var6) {
+        removeBlock = true;
         if (!var1.isStatic) {
             if ((var5 & 1) == 1) {
                 EntityTNTPrimed var7 = new EntityTNTPrimed(var1, (double)((float)var2 + 0.5F), (double)((float)var3 + 0.5F), (double)((float)var4 + 0.5F), var6);
-                var1.addEntity(var7);
-                var1.makeSound(var7, "game.tnt.primed", 1.0F, 1.0F);
+
+                // CraftBukkit start - fire ExplosionPrimeEvent
+                org.bukkit.event.entity.ExplosionPrimeEvent event = new org.bukkit.event.entity.ExplosionPrimeEvent((org.bukkit.entity.Explosive) var7.getBukkitEntity());
+                var1.getServer().getPluginManager().callEvent(event);
+
+                if (event.isCancelled()) {
+                    removeBlock = false;
+                } else {
+                    var1.addEntity(var7);
+                    var1.makeSound(var7, "game.tnt.primed", 1.0F, 1.0F);
+                }
+                // CraftBukkit end
             }
         }
     }
@@ -65,7 +84,11 @@ public class BlockTNT extends Block {
     public boolean interact(World var1, int var2, int var3, int var4, EntityHuman var5, int var6, float var7, float var8, float var9) {
         if (var5.bD() != null && var5.bD().getItem() == Items.FLINT_AND_STEEL) {
             this.a(var1, var2, var3, var4, 1, var5);
-            var1.setAir(var2, var3, var4);
+            // CraftBukkit start - don't remove block if TNT priming is cancelled
+            if (removeBlock) {
+                var1.setAir(var2, var3, var4);
+            }
+            // CraftBukkit end
             var5.bD().damage(1, var5);
             return true;
         } else {
diff --git a/src/main/java/net/minecraft/server/EntityCreeper.java b/src/main/java/net/minecraft/server/EntityCreeper.java
index 3c51de5..7912a72 100644
--- a/src/main/java/net/minecraft/server/EntityCreeper.java
+++ b/src/main/java/net/minecraft/server/EntityCreeper.java
@@ -12,6 +12,7 @@ public class EntityCreeper extends EntityMonster {
     private int maxFuseTicks = 30;
     private int explosionRadius = 3;
     private int record = -1; // CraftBukkit
+    private ExplosionPrimeEvent primeEvent; // CraftBukkit
 
     public EntityCreeper(World world) {
         super(world);
@@ -91,7 +92,17 @@ public class EntityCreeper extends EntityMonster {
             int i = this.bZ();
 
             if (i > 0 && this.fuseTicks == 0) {
-                this.makeSound("creeper.primed", 1.0F, 0.5F);
+                // CraftBukkit start - Fire ExplosionPrimeEvent
+                float radius = this.isPowered() ? 2 * explosionRadius : explosionRadius;
+
+                primeEvent = new ExplosionPrimeEvent(this.getBukkitEntity(), radius, false);
+                this.world.getServer().getPluginManager().callEvent(primeEvent);
+                if (primeEvent.isCancelled()) {
+                    i = 0;
+                } else {
+                    this.makeSound("creeper.primed", 1.0F, 0.5F);
+                }
+                // CraftBukkit end
             }
 
             this.fuseTicks += i;
@@ -217,19 +228,8 @@ public class EntityCreeper extends EntityMonster {
     private void cc() {
         if (!this.world.isStatic) {
             boolean flag = this.world.getGameRules().getBoolean("mobGriefing");
-
-            // CraftBukkit start
-            float radius = this.isPowered() ? 6.0F : 3.0F;
-
-            ExplosionPrimeEvent event = new ExplosionPrimeEvent(this.getBukkitEntity(), radius, false);
-            this.world.getServer().getPluginManager().callEvent(event);
-            if (!event.isCancelled()) {
-                this.world.createExplosion(this, this.locX, this.locY, this.locZ, event.getRadius(), event.getFire(), flag);
-                this.die();
-            } else {
-                this.fuseTicks = 0;
-            }
-            // CraftBukkit end
+            this.world.createExplosion(this, this.locX, this.locY, this.locZ, primeEvent.getRadius(), primeEvent.getFire(), flag); // CraftBukkit - Use stored explosion parameters
+            this.die();
         }
     }
 
diff --git a/src/main/java/net/minecraft/server/EntityLargeFireball.java b/src/main/java/net/minecraft/server/EntityLargeFireball.java
index d136d2d..b3ab7ee 100644
--- a/src/main/java/net/minecraft/server/EntityLargeFireball.java
+++ b/src/main/java/net/minecraft/server/EntityLargeFireball.java
@@ -20,16 +20,8 @@ public class EntityLargeFireball extends EntityFireball {
                 movingobjectposition.entity.damageEntity(DamageSource.fireball(this, this.shooter), 6.0F);
             }
 
-            // CraftBukkit start
-            ExplosionPrimeEvent event = new ExplosionPrimeEvent((org.bukkit.entity.Explosive) org.bukkit.craftbukkit.entity.CraftEntity.getEntity(this.world.getServer(), this));
-            this.world.getServer().getPluginManager().callEvent(event);
-
-            if (!event.isCancelled()) {
-                // give 'this' instead of (Entity) null so we know what causes the damage
-                this.world.createExplosion(this, this.locX, this.locY, this.locZ, event.getRadius(), event.getFire(), this.world.getGameRules().getBoolean("mobGriefing"));
-            }
-            // CraftBukkit end
-
+            // CraftBukkit - give 'this' instead of (Entity) null so we know what causes the damage
+            this.world.createExplosion(this, this.locX, this.locY, this.locZ, yield, isIncendiary, this.world.getGameRules().getBoolean("mobGriefing"));
             this.die();
         }
     }
diff --git a/src/main/java/net/minecraft/server/EntityTNTPrimed.java b/src/main/java/net/minecraft/server/EntityTNTPrimed.java
index 92f0042..323477d 100644
--- a/src/main/java/net/minecraft/server/EntityTNTPrimed.java
+++ b/src/main/java/net/minecraft/server/EntityTNTPrimed.java
@@ -70,17 +70,8 @@ public class EntityTNTPrimed extends Entity {
 
     private void explode() {
         // CraftBukkit start
-        // float f = 4.0F;
-
-        org.bukkit.craftbukkit.CraftServer server = this.world.getServer();
-
-        ExplosionPrimeEvent event = new ExplosionPrimeEvent((org.bukkit.entity.Explosive) org.bukkit.craftbukkit.entity.CraftEntity.getEntity(server, this));
-        server.getPluginManager().callEvent(event);
-
-        if (!event.isCancelled()) {
-            // give 'this' instead of (Entity) null so we know what causes the damage
-            this.world.createExplosion(this, this.locX, this.locY, this.locZ, event.getRadius(), event.getFire(), true);
-        }
+        org.bukkit.entity.Explosive explosive = (org.bukkit.entity.Explosive) this.getBukkitEntity();
+        this.world.createExplosion(this, this.locX, this.locY, this.locZ, explosive.getYield(), explosive.isIncendiary(), true);
         // CraftBukkit end
     }
 
-- 
1.8.3.2

