From 6dde9276511c8f366f1bb474c6298cde33066f10 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Wed, 15 May 2013 20:09:16 -0500
Subject: [PATCH] Fire BlockBreakEvent for extinguishing fire and fix
 cancelling interact events working with fire


diff --git a/src/main/java/net/minecraft/server/PlayerInteractManager.java b/src/main/java/net/minecraft/server/PlayerInteractManager.java
index bceba7b..1e5f9af 100644
--- a/src/main/java/net/minecraft/server/PlayerInteractManager.java
+++ b/src/main/java/net/minecraft/server/PlayerInteractManager.java
@@ -104,6 +104,13 @@ public class PlayerInteractManager {
         if (!this.gamemode.isAdventure() || this.player.d(i, j, k)) {
             if (event.isCancelled()) {
                 // Let the client know the block still exists
+                // Check to see if fire was doused, if so, update the fire instead of the block the fire is on
+                org.bukkit.block.Block block = this.world.getWorld().getBlockAt(i, j, k).getRelative(org.bukkit.craftbukkit.block.CraftBlock.notchToBlockFace(l));
+                if (block.getTypeId() == Block.FIRE.id) {
+                    i = block.getX();
+                    j = block.getY();
+                    k = block.getZ();
+                }
                 ((EntityPlayer) this.player).playerConnection.sendPacket(new PacketPlayOutBlockChange(i, j, k, this.world));
                 // Update any tile entity data for this block
                 TileEntity tileentity = this.world.getTileEntity(i, j, k);
@@ -114,7 +121,7 @@ public class PlayerInteractManager {
             }
             // CraftBukkit end
             if (this.isCreative()) {
-                if (!this.world.douseFire((EntityHuman) null, i, j, k, l)) {
+                if (!this.world.douseFire(this.player, i, j, k, l)) { // CraftBukkit - add this.player
                     this.breakBlock(i, j, k);
                 }
             } else {
@@ -137,7 +144,7 @@ public class PlayerInteractManager {
                     block.attack(this.world, i, j, k, this.player);
                     f = block.getDamage(this.player, this.player.world, i, j, k);
                     // Allow fire punching to be blocked
-                    this.world.douseFire((EntityHuman) null, i, j, k, l);
+                    this.world.douseFire(this.player, i, j, k, l); // CraftBukkit - add this.player
                 }
 
                 if (event.useItemInHand() == Event.Result.DENY) {
diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index b48c1c1..9b8dc95 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1668,8 +1668,13 @@ public abstract class World implements IBlockAccess {
         }
 
         if (this.getType(i, j, k) == Blocks.FIRE) {
-            this.a(entityhuman, 1004, i, j, k, 0);
-            this.setAir(i, j, k);
+            // CraftBukkit start
+            if (entityhuman instanceof EntityPlayer) {
+                if (((EntityPlayer) entityhuman).playerInteractManager.breakBlock(i, j, k)) {
+                    this.triggerEffect(1004, i, j, k, 0);
+                }
+            }
+            // CraftBukkit end
             return true;
         } else {
             return false;
-- 
1.8.3.2

