From 1b44098f2d136f1565ad25ca1c312e8472c4567e Mon Sep 17 00:00:00 2001
From: Marcos Vives Del Sol <socram8888@gmail.com>
Date: Fri, 13 Jun 2014 16:45:51 +0200
Subject: [PATCH] Add missing effects


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftEffect.java b/src/main/java/org/bukkit/craftbukkit/CraftEffect.java
index 7de0de5..c610812 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftEffect.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftEffect.java
@@ -55,6 +55,9 @@ public class CraftEffect {
             Validate.isTrue(((Material) data).isBlock(), "Material is not a block!");
             datavalue = ((Material) data).getId();
             break;
+        case LAND_DUST:
+            datavalue = (Integer) data;
+            break;
         default:
             datavalue = 0;
         }
-- 
1.7.9

