From e98985c7be4893f33b25644f9bbbdd0a375e5421 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Tue, 23 Apr 2013 19:45:55 -0500
Subject: [PATCH] Call AsyncPlayerPreLoginEvent regardless of online-mode


diff --git a/src/main/java/net/minecraft/server/LoginListener.java b/src/main/java/net/minecraft/server/LoginListener.java
index 24a77a2..5825a7f 100644
--- a/src/main/java/net/minecraft/server/LoginListener.java
+++ b/src/main/java/net/minecraft/server/LoginListener.java
@@ -106,7 +106,8 @@ public class LoginListener implements PacketLoginInListener {
             this.g = EnumProtocolState.KEY;
             this.networkManager.handle(new PacketLoginOutEncryptionBegin(this.j, this.server.I().getPublic(), this.e), new GenericFutureListener[0]);
         } else {
-            this.g = EnumProtocolState.READY_TO_ACCEPT;
+            // CraftBukkit - call async login event no matter what
+            (new ThreadPlayerLookupUUID(this, "User Authenticator #" + b.incrementAndGet())).start();
         }
     }
 
diff --git a/src/main/java/net/minecraft/server/ThreadPlayerLookupUUID.java b/src/main/java/net/minecraft/server/ThreadPlayerLookupUUID.java
index 09580d7..bbaded5 100644
--- a/src/main/java/net/minecraft/server/ThreadPlayerLookupUUID.java
+++ b/src/main/java/net/minecraft/server/ThreadPlayerLookupUUID.java
@@ -22,15 +22,20 @@ class ThreadPlayerLookupUUID extends Thread {
     }
 
     public void run() {
-        boolean allowed = false; // CraftBukkit
-        try {
-            String s = (new BigInteger(MinecraftEncryption.a(LoginListener.a(this.a), LoginListener.b(this.a).I().getPublic(), LoginListener.c(this.a)))).toString(16);
+        // CraftBukkit start
+        boolean allowed = false;
 
-            LoginListener.a(this.a, LoginListener.b(this.a).as().hasJoinedServer(new GameProfile((String) null, LoginListener.d(this.a).getName()), s));
-        } catch(Exception exception) {}
+        if (this.server.getOnlineMode()) {
+            try {
+                String s = (new BigInteger(MinecraftEncryption.a(LoginListener.a(this.a), LoginListener.b(this.a).I().getPublic(), LoginListener.c(this.a)))).toString(16);
 
-        // CraftBukkit start
-        allowed = LoginListener.d(this.a) != null;
+                LoginListener.a(this.a, LoginListener.b(this.a).as().hasJoinedServer(new GameProfile((String) null, LoginListener.d(this.a).getName()), s));
+            } catch(Exception exception) {}
+
+            allowed = LoginListener.d(this.a) != null;
+        } else {
+            allowed = true;
+        }
 
         if (!this.a.networkManager.d()) {
             return;
-- 
1.8.5.1

