From 3a9f454f48842f31a4bb3aa70800e1a8e3440244 Mon Sep 17 00:00:00 2001
From: Steve Anton <anxuiz.nx@gmail.com>
Date: Sat, 4 Aug 2012 10:55:59 -0700
Subject: [PATCH] Implement the overhead name API.


diff --git a/src/main/java/net/minecraft/server/PacketPlayOutNamedEntitySpawn.java b/src/main/java/net/minecraft/server/PacketPlayOutNamedEntitySpawn.java
index 8bab528..b663823 100644
--- a/src/main/java/net/minecraft/server/PacketPlayOutNamedEntitySpawn.java
+++ b/src/main/java/net/minecraft/server/PacketPlayOutNamedEntitySpawn.java
@@ -23,6 +23,7 @@ public class PacketPlayOutNamedEntitySpawn extends Packet {
 
     public PacketPlayOutNamedEntitySpawn(EntityHuman entityhuman) {
         this.a = entityhuman.getId();
+        // CraftBukkit - NOTE: Patch broken
         this.b = entityhuman.getProfile();
         this.c = MathHelper.floor(entityhuman.locX * 32.0D);
         this.d = MathHelper.floor(entityhuman.locY * 32.0D);
-- 
1.8.5.1

