From 40c3cf05e06c0ac84aa3fa3b8f2ef35ec7bbdc0c Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Sat, 17 Nov 2012 12:11:38 -0600
Subject: [PATCH] Add the PlayerHandshakeEvent


diff --git a/src/main/java/net/minecraft/server/HandshakeListener.java b/src/main/java/net/minecraft/server/HandshakeListener.java
index fa13c5b..5c8f4d3 100644
--- a/src/main/java/net/minecraft/server/HandshakeListener.java
+++ b/src/main/java/net/minecraft/server/HandshakeListener.java
@@ -89,6 +89,7 @@ public class HandshakeListener implements PacketHandshakingInListener {
                 }
                 // CraftBukkit End
                 ((LoginListener) this.b.getPacketListener()).hostname = packethandshakinginsetprotocol.b + ":" + packethandshakinginsetprotocol.c; // CraftBukkit - set hostname
+                // TODO - fire handshake event
             }
             break;
 
-- 
1.8.5.1

