From 8ef95354382a0fe07d23c479b59dadc233736c6e Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Sun, 25 Nov 2012 13:56:45 -0600
Subject: [PATCH] Remove the entity destroy queue


diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
index ea3c74f..1507cb7 100644
--- a/src/main/java/net/minecraft/server/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/EntityPlayer.java
@@ -37,7 +37,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
     public double d;
     public double e;
     public final List chunkCoordIntPairQueue = new LinkedList();
-    public final List removeQueue = new LinkedList();
+    // public final List removeQueue = new LinkedList(); CraftBukkit - remove the removeQueue
     private final ServerStatisticManager bO;
     private float bP = Float.MIN_VALUE;
     private float bQ = -1.0E8F;
@@ -187,6 +187,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
             this.activeContainer = this.defaultContainer;
         }
 
+        /* CraftBukkit - remove the removeQueue
         while (!this.removeQueue.isEmpty()) {
             int i = Math.min(this.removeQueue.size(), 127);
             int[] aint = new int[i];
@@ -200,6 +201,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
 
             this.playerConnection.sendPacket(new PacketPlayOutEntityDestroy(aint));
         }
+        */
 
         if (!this.chunkCoordIntPairQueue.isEmpty()) {
             ArrayList arraylist = new ArrayList();
@@ -872,7 +874,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
         this.lastSentExp = -1;
         this.bQ = -1.0F;
         this.bR = -1;
-        this.removeQueue.addAll(((EntityPlayer) entityhuman).removeQueue);
+        // this.removeQueue.addAll(((EntityPlayer) entityhuman).removeQueue);  CraftBukkit - remove the removeQueue
     }
 
     protected void a(MobEffect mobeffect) {
diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index 748417d..1fc9fee 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -342,18 +342,12 @@ public class EntityTrackerEntry {
     // CraftBukkit end
 
     public void a() {
-        Iterator iterator = this.trackedPlayers.iterator();
-
-        while (iterator.hasNext()) {
-            EntityPlayer entityplayer = (EntityPlayer) iterator.next();
-
-            entityplayer.removeQueue.add(Integer.valueOf(this.tracker.getId()));
-        }
+        this.broadcast(new PacketPlayOutEntityDestroy(this.tracker.getId())); // CraftBukkit - remove destroy queue
     }
 
     public void a(EntityPlayer entityplayer) {
         if (this.trackedPlayers.contains(entityplayer)) {
-            entityplayer.removeQueue.add(Integer.valueOf(this.tracker.getId()));
+            entityplayer.playerConnection.sendPacket(new PacketPlayOutEntityDestroy(this.tracker.getId())); // CraftBukkit - remove destroy queue
             this.trackedPlayers.remove(entityplayer);
         }
     }
@@ -373,7 +367,7 @@ public class EntityTrackerEntry {
                         }
                     }
 
-                    entityplayer.removeQueue.remove(Integer.valueOf(this.tracker.getId()));
+                    // entityplayer.removeQueue.remove(Integer.valueOf(this.tracker.getId())); CraftBukkit - remove destroy queue
                     // CraftBukkit end
 
                     this.trackedPlayers.add(entityplayer);
@@ -457,7 +451,7 @@ public class EntityTrackerEntry {
                 }
             } else if (this.trackedPlayers.contains(entityplayer)) {
                 this.trackedPlayers.remove(entityplayer);
-                entityplayer.removeQueue.add(Integer.valueOf(this.tracker.getId()));
+                entityplayer.playerConnection.sendPacket(new PacketPlayOutEntityDestroy(this.tracker.getId())); // CraftBukkit - remove destroy queue
             }
         }
     }
@@ -579,7 +573,11 @@ public class EntityTrackerEntry {
     public void clear(EntityPlayer entityplayer) {
         if (this.trackedPlayers.contains(entityplayer)) {
             this.trackedPlayers.remove(entityplayer);
+<<<<<<< HEAD
             entityplayer.removeQueue.add(Integer.valueOf(this.tracker.getId()));
+=======
+            entityplayer.playerConnection.sendPacket(new Packet29DestroyEntity(this.tracker.id));
+>>>>>>> Remove the entity destroy queue
         }
     }
 }
-- 
1.8.5.1

