From d6192eee52c1db435674b81f1be3e3ca576f18b6 Mon Sep 17 00:00:00 2001
From: Daniel <ZetaetaDaniel@gmail.com>
Date: Sat, 12 May 2012 16:07:34 +0100
Subject: [PATCH] Adds ability to set the number of online players sent in
 ServerListPingEvent


diff --git a/src/main/java/net/minecraft/server/PacketStatusListener.java b/src/main/java/net/minecraft/server/PacketStatusListener.java
index 7903c43..6209d10 100644
--- a/src/main/java/net/minecraft/server/PacketStatusListener.java
+++ b/src/main/java/net/minecraft/server/PacketStatusListener.java
@@ -109,7 +109,7 @@ public class PacketStatusListener implements PacketStatusInListener {
             }
         }
 
-        ServerPingPlayerSample playerSample = new ServerPingPlayerSample(event.getMaxPlayers(), profiles.size());
+        ServerPingPlayerSample playerSample = new ServerPingPlayerSample(event.getMaxPlayers(), event.getNumPlayers());
         playerSample.a(profiles.toArray(new GameProfile[profiles.size()]));
 
         ServerPing ping = new ServerPing();
-- 
1.8.5.1

