From 9b874da2ae283073fe6ae2d6a8aec34061f6cb32 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Tue, 9 Jul 2013 17:06:59 -0500
Subject: [PATCH] Fix invalid positions kicking players


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 61f4cc4..b5c0e90 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -222,6 +222,11 @@ public class PlayerConnection implements PacketPlayInListener {
             double delta = Math.pow(this.lastPosX - to.getX(), 2) + Math.pow(this.lastPosY - to.getY(), 2) + Math.pow(this.lastPosZ - to.getZ(), 2);
             float deltaAngle = Math.abs(this.lastYaw - to.getYaw()) + Math.abs(this.lastPitch - to.getPitch());
 
+            if (Double.isNaN(packet10flying.x) || Double.isNaN(packet10flying.y) || Double.isNaN(packet10flying.z) || Double.isNaN(packet10flying.stance) || (packet10flying.hasLook && (Double.isNaN(packet10flying.yaw) || Double.isNaN(packet10flying.pitch)))) {
+                player.teleport(to, PlayerTeleportEvent.TeleportCause.UNKNOWN);
+                return;
+            }
+
             if ((delta > 1f / 256 || deltaAngle > 10f) && (this.checkMovement && !this.player.dead)) {
                 this.lastPosX = to.getX();
                 this.lastPosY = to.getY();
@@ -461,6 +466,18 @@ public class PlayerConnection implements PacketPlayInListener {
             f1 = 0;
         }
 
+        if(Double.isNaN(d0)) {
+            d0 = this.y;
+        }
+
+        if(Double.isNaN(d1)) {
+            d1 = this.z;
+        }
+
+        if(Double.isNaN(d2)) {
+            d2 = this.p;
+        }
+
         this.lastPosX = d0;
         this.lastPosY = d1;
         this.lastPosZ = d2;
-- 
1.8.5.1

