From 8836e45cdb86cb104b8b1bdac3836223f0b8af2b Mon Sep 17 00:00:00 2001
From: Marcos Vives Del Sol <socram8888@gmail.com>
Date: Sun, 28 Jul 2013 22:10:42 +0200
Subject: [PATCH] Add hostname to PlayerHandshakeEvent


diff --git a/src/main/java/net/minecraft/server/HandshakeListener.java b/src/main/java/net/minecraft/server/HandshakeListener.java
index 5c8f4d3..bbef73a 100644
--- a/src/main/java/net/minecraft/server/HandshakeListener.java
+++ b/src/main/java/net/minecraft/server/HandshakeListener.java
@@ -90,6 +90,7 @@ public class HandshakeListener implements PacketHandshakingInListener {
                 // CraftBukkit End
                 ((LoginListener) this.b.getPacketListener()).hostname = packethandshakinginsetprotocol.b + ":" + packethandshakinginsetprotocol.c; // CraftBukkit - set hostname
                 // TODO - fire handshake event
+                // TODO - add hostname to handshake event
             }
             break;
 
-- 
1.8.5.1

