From 7813027267efe02d5dc80a77cff56d1ae1918fc0 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Tue, 1 Oct 2013 21:10:50 -0500
Subject: [PATCH] Fix clearing potion effects on respawn -- missed rename


diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
index 0e408e1..e03464c 100644
--- a/src/main/java/net/minecraft/server/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/EntityPlayer.java
@@ -1053,7 +1053,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
         this.expTotal = this.newTotalExp;
         this.exp = 0;
         this.deathTicks = 0;
-        this.aP(); // Should be remapped: removeAllEffects should be remapped to this.
+        this.aP(); // Should be remapped: removeAllEffects should be remapped to this...
         this.updateEffects = true;
         this.activeContainer = this.defaultContainer;
         this.killer = null;
-- 
1.8.5.1

