From c837e185635228db524d837ceb0d45383e0d0d6f Mon Sep 17 00:00:00 2001
From: ShinyDialga45 <shinydialga45@gmail.com>
Date: Sun, 30 Nov 2014 10:53:44 -0600
Subject: [PATCH] Fix leash interaction bugs


diff --git a/src/main/java/net/minecraft/server/PlayerInteractManager.java b/src/main/java/net/minecraft/server/PlayerInteractManager.java
index 325b87e..261b69c 100644
--- a/src/main/java/net/minecraft/server/PlayerInteractManager.java
+++ b/src/main/java/net/minecraft/server/PlayerInteractManager.java
@@ -397,6 +397,19 @@ public class PlayerInteractManager {
                 result = (event.useItemInHand() != Event.Result.ALLOW);
             } else if (!entityhuman.isSneaking() || itemstack == null) {
                 result = block.interact(world, i, j, k, entityhuman, l, f, f1, f2);
+                ItemStack heldItem = entityhuman.be();
+                //Checks if the player has a leash attached, if not it checks if they clicked a fence and their item is banned to prevent a glitch.
+                if (heldItem != null && heldItem.getItem() == Items.LEASH && this.world.isStatic) {
+                    result = false;
+                } else if (block == Blocks.FENCE || block == Blocks.NETHER_FENCE) {
+                    Item[] bannedItems = {Items.BOW, Items.WOOD_SWORD, Items.GOLD_SWORD, Items.STONE_SWORD,
+                            Items.IRON_SWORD, Items.DIAMOND_SWORD};
+                    for (Item item : bannedItems) {
+                        if (itemstack != null && itemstack.getItem() == item) {
+                            return result;
+                        }
+                    }
+                }
             }
 
             if (itemstack != null && !result) {
-- 
1.9.4.msysgit.2

