From 67e8798e3d432c4a02633961384274ca32a53883 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Sun, 16 Jun 2013 16:57:59 -0500
Subject: [PATCH] Add getting and setting of entity count


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index a90b390..94d5000 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -666,6 +666,20 @@ public final class Bukkit {
     }
 
     /**
+     * @see Server#getEntityCount()
+     */
+    public static int getEntityCount() {
+        return server.getEntityCount();
+    }
+
+    /**
+     * @see Server#setEntityCount()
+     */
+    public static void setEntityCount(int count) {
+        server.setEntityCount(count);
+    }
+
+    /**
      * @see Server#getServerIcon()
      */
     public static CachedServerIcon getServerIcon() {
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 3b6ea63..4aa78a7 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -752,6 +752,20 @@ public interface Server extends PluginMessageRecipient {
     ScoreboardManager getScoreboardManager();
 
     /**
+     * Gets the current entity count
+     *
+     * @return current entity count
+     */
+    int getEntityCount();
+
+    /**
+     * Sets the current entity count
+     *
+     * @param count The new entity count
+     */
+    void setEntityCount(int count);
+
+    /**
      * Gets an instance of the server's default server-icon.
      *
      * @return the default server-icon; null values may be used by the
-- 
1.8.5.1

