From f0a32bb34e4258918a6fbcf911060ba09c7d74c8 Mon Sep 17 00:00:00 2001
From: Chad Waters <authorblues@gmail.com>
Date: Wed, 6 Feb 2013 14:42:01 -0500
Subject: [PATCH] Add safe methods from @md-5's offlineUtil


diff --git a/src/main/java/org/bukkit/OfflinePlayer.java b/src/main/java/org/bukkit/OfflinePlayer.java
index 6196459..8caae04 100644
--- a/src/main/java/org/bukkit/OfflinePlayer.java
+++ b/src/main/java/org/bukkit/OfflinePlayer.java
@@ -3,6 +3,7 @@ package org.bukkit;
 import org.bukkit.configuration.serialization.ConfigurationSerializable;
 import org.bukkit.entity.AnimalTamer;
 import org.bukkit.entity.Player;
+import org.bukkit.inventory.PlayerInventory;
 import org.bukkit.permissions.ServerOperator;
 
 public interface OfflinePlayer extends ServerOperator, AnimalTamer, ConfigurationSerializable {
@@ -100,4 +101,18 @@ public interface OfflinePlayer extends ServerOperator, AnimalTamer, Configuratio
      */
     public Location getBedSpawnLocation();
 
+    /**
+     * Get the player's inventory. This inventory will not save upon modification, it must be saved using the saveInventory() method.
+     * This inventory will also have a null owner.
+     *
+     * @return The inventory of the player, this also contains the armor slots.
+     */
+    public PlayerInventory getInventory();
+
+     /**
+     * Gets the last Location where the player was before he logged out of the server.
+     *
+     * @return Last Location of the player.
+     */
+    public Location getLocation();
 }
-- 
1.8.5.1

