From 2605cfae6fb397196515184261d1e68d38e605ec Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Wed, 29 May 2013 23:52:37 -0500
Subject: [PATCH] Visibility API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 96349af..4b025b4 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -839,6 +839,13 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     public void setAllowFlight(boolean flight);
 
     /**
+     * Hides an entity from this player
+     *
+     * @param entity Entity to hide
+     */
+    public void hide(Entity entity);
+
+    /**
      * Hides a player from this player
      *
      * @param player Player to hide
@@ -846,6 +853,13 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     public void hidePlayer(Player player);
 
     /**
+     * Allows this player to see an entity that was previously hidden
+     *
+     * @param entity Entity to show
+     */
+    public void show(Entity entity);
+
+    /**
      * Allows this player to see a player that was previously hidden
      *
      * @param player Player to show
@@ -853,12 +867,14 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     public void showPlayer(Player player);
 
     /**
-     * Checks to see if a player has been hidden from this player
+     * Checks to see if an entity has been hidden from this player
      *
-     * @param player Player to check
-     * @return True if the provided player is not being hidden from this
+     * @param entity Entity to check
+     * @return True if the provided entity is not being hidden from this
      *     player
      */
+    public boolean canSee(Entity entity);
+
     public boolean canSee(Player player);
 
     /**
-- 
1.8.3.2

