From ba8eab22bd884ab994b7e7a94b2b75e4e1a3daea Mon Sep 17 00:00:00 2001
From: Isaac Moore <rmsy@me.com>
Date: Sun, 5 Jan 2014 20:32:00 -0600
Subject: [PATCH] Allow plugins to access player locale


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 549c07b..cb4a96c 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1087,4 +1087,11 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      * @return The hostname and port
      */
     public String getHostname();
+
+    /**
+     * Gets the player's client-determined, ISO-standard locale.
+     *
+     * @return The locale
+     */
+    public String getLocale();
  }
-- 
1.7.9

