From 2276ab8bc9036580e2405d609a124b020e9654ef Mon Sep 17 00:00:00 2001
From: Isaac Moore <rmsy@me.com>
Date: Sun, 5 Jan 2014 20:32:00 -0600
Subject: [PATCH] Allow plugins to access player locale


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 4096f97..a8e9cf3 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1091,4 +1091,11 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      * @return The hostname and port
      */
     public String getHostname();
+
+    /**
+     * Gets the player's client-determined, ISO-standard locale.
+     *
+     * @return The locale
+     */
+    public String getLocale();
  }
-- 
1.8.3.2

